src/slash_commands/score.rs

```diff
@@ -0,0 +1,139 @@
+
+const DEFAULT_MODE: &str = "default";
+const LOOSE_MODE: &str = "loose";
+const UNLIMITED_MODE: &str = "unlimited";
+
+const MAX_HAN: i64 = 1600;
+const MAX_FU: i64 = 100000;
+
+pub struct ScoreCommand;
+
+impl ScoreCommand {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait]
+impl SlashCommand for ScoreCommand {
+    fn get_name(&self) -> &'static str {
+        HAND_COMMAND
+    }
+
+    fn add_application_command(&self, command: &mut CreateApplicationCommand) {
+        command
+            .description("Calculate the score for given number of han and fu points")
+            .create_option(|option| {
+                option
+                    .name(HAN_OPTION)
+                    .description("Number of han points")
+                    .kind(CommandOptionType::Integer)
+                    .max_int_value(MAX_HAN)
+                    .required(true)
+            })
+            .create_option(|option| {
+                option
+                    .name(FU_OPTION)
+                    .description("Number of fu points")
+                    .kind(CommandOptionType::Integer)
+                    .max_int_value(MAX_FU)
+                    .required(true)
+            })
+            .create_option(|option| {
+                option
+                    .name(MODE_OPTION)
+                    .description("Calculating mode")
+                    .kind(CommandOptionType::String)
+                    .add_string_choice("Default", DEFAULT_MODE)
+                    .add_string_choice("Loose", LOOSE_MODE)
+                    .add_string_choice("Unlimited", UNLIMITED_MODE)
+                    .required(false)
+            });
+    }
+
+    async fn handle(
+        &self,
+        ctx: &Context,
+        command: &ApplicationCommandInteraction,
+        _chombot: &Chombot,
+    ) -> SlashCommandResult {
+        let han = get_int_option(&command.data.options, HAN_OPTION).ok_or("Missing han value")?;
+        let fu = get_int_option(&command.data.options, FU_OPTION).ok_or("Missing fu value")?;
+        let mode = get_string_option(&command.data.options, MODE_OPTION).unwrap_or(DEFAULT_MODE);
+        let points_calculation_mode = match mode {
+            DEFAULT_MODE => Ok(PointsCalculationMode::Default),
+            LOOSE_MODE => Ok(PointsCalculationMode::Loose),
+            UNLIMITED_MODE => Ok(PointsCalculationMode::Unlimited),
+            _ => Err(format!("Invalid mode: {mode}")),
+        }?;
+
+        let han = Han::new(u32::try_from(han)?);
+        let fu = Fu::new(u32::try_from(fu)?);
+        let points = PointsCustom::<BigUint>::from_calculated(points_calculation_mode, han, fu)?;
+        let embed = create_points_embed(han, fu, &points)?;
+
+        command
+            .edit_original_interaction_response(&ctx.http, |response| response.add_embed(embed))
+            .await?;
+
+        Ok(())
+    }
+}
+
+fn create_points_embed(
+    han: Han,
+    fu: Fu,
+    points: &PointsCustom<BigUint>,
```

Comment:
nit again, I don't even know if it's worth it but at least would be consistent with the rest being passed by value
```suggestion
    points: PointsCustom<BigUint>,
```

Reply:
oh and also I suppose we could introduce a typealias for `PointsCustom<BigUint>` and reuse it in L91

Reply:
The others are passed by value since they implement `Copy` by being merely a wrapped `u32`. `BigUint` doesn't (yet alone `PointsCustom<BigUint>`), so I'd prefer to leave it as is.

Reply:
> oh and also I suppose we could introduce a typealias for `PointsCustom<BigUint>` and reuse it in L91

Not a bad idea, will do.

Reply:
Added the type alias.